socless/integrations.py

```diff
@@ -235,5 +242,7 @@ def execute(self):
             raise Exception("Result returned from the integration handler is not a Python dictionary. Must be a Python dictionary")
 
         if not self.testing:
-            self.execution_context.save_state_results(self.state_name,result)
+            if 'errors' not in self.context:
+                self.context['errors'] = {}
+            self.execution_context.save_state_results(self.state_name,result, errors=self.context['errors'])
```

Comment:
Seems like save_state_results already knows how to handle a situation where `errors = {}`?

If so, a cleaner implementation of these three lines may be:
```
self.execution_context.save_state_results(self.state_name, result, errors=self.context.get('errors', {}))
```

`self.context.get('errors', {})` attempts to fetch `errors` from self.context, and defaults to {} if `errors` not in self.context.

The dictionary .get method is a concise way to fetch a value if a key exists but default to another value if the key doesn't